R/lint.R

```diff
@@ -63,6 +63,32 @@ lint <- function(filename, linters = NULL, cache = FALSE, ..., parse_settings =
     names(linters) <- auto_names(linters)
   }
 
+  linters <- Map(
+    function(obj, name) {
+      if (inherits(obj, "linter")) {
+      } else if (inherits(obj, "function")) {
+        if (is.null(formals(obj))) {
+          old <- "Passing linters as variables"
+          new <- "a call to the linters (see ?linters)"
+          lintr_deprecated(old = old, new = new, version = "2.0.1.9001",
+                           type = "")
+          obj <- obj()
+        } else {
```

Comment:
Should we be stricter here and check that `formals(obj)` has exactly one argument, otherwise stop?

Reply:
Testing this now via `obj <- Linter(obj)`